multi-ai-cli/core/cache.py

```
@@ -3,21 +3,58 @@
 from typing import List, Dict, Optional
 
 CACHE_DIR = Path.home() / ".multi-ai-cache"
-CACHE_DIR.mkdir(exist_ok=True)
+CACHE_DIR.mkdir(parents=True, exist_ok=True)
+if not CACHE_DIR.is_symlink():
+    try:
+        CACHE_DIR.chmod(0o700)
+    except Exception:
```

Pylint (4.0.7)

[warning] Catching too general exception Exception

(W0718)